sys/shell/cmds/iw.c

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting Gmbh
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_shell_commands
+ * @{
+ *
+ * @file
+ * @brief       Shell commands for interacting with Wi-Fi interfaces
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "kernel_defines.h"
+#include "mutex.h"
+#include "net/l2scan_list.h"
+#include "net/netdev/wifi.h"
+#include "net/netopt.h"
+#include "net/wifi.h"
+#include "sched.h"
+#include "shell.h"
+#include "net/netif.h"
+#include "net/netdev.h"
+#include "fmt.h"
+#include "string_utils.h"
+
+#define SC_IW_AP_NUMOF                  10
+
+static mutex_t _sync = MUTEX_INIT;
+
+static struct {
+    unsigned numof;
+    wifi_scan_result_t ap[SC_IW_AP_NUMOF];
+} _aps;
+
+static union {
+    wifi_security_wep_psk_t wep;
+    wifi_security_wpa_psk_t wpa;
+    wifi_security_eap_user_t user;
+} _cred;
+
+static const wifi_scan_result_t *_get_ap(const char *ssid)
+{
+    for (unsigned i = 0; i < _aps.numof; i++) {
+        if (!strcmp(_aps.ap[i].ssid, ssid)) {
+            return &_aps.ap[i];
+        }
+    }
+    return NULL;
+}
+
+static const char *_ssec(wifi_security_mode_t mode)
+{
+    switch (mode) {
+        case WIFI_SECURITY_MODE_NONE:
+            return "open";
+        case WIFI_SECURITY_MODE_WEP_PSK:
+            return "WEP";
+        case WIFI_SECURITY_MODE_WPA_PSK:
+            return "WPA";
+        case WIFI_SECURITY_MODE_EAP_USER:
+            return "EAP username";
+        default:
+            return "unknown";
+    }
+}
+
+static void _list_ap(void)
+{
+    puts("                            SSID |         SEC |  RSSI | CHANNEL");
```

Comment:
Why are SSID and SEC column right aligned?

Reply:
There was no particular reason for that it is just what e.g. `%32s` does. Do you want me to left-align the columns with `%-32s`?

Reply:
It looks a bit strange, should be right aligned but names/labels a.s.o should be left aligned. Yes, please change it.